var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -355,9 +369,41 @@ def nvariant_bool(feature):
         else:
             cmake_args.append('-DVTKm_ENABLE_CUDA:BOOL=%s' %
                               variant_bool('+cuda'))
-        if spec.satisfies('+cuda') and not spec.satisfies('cuda_arch=native'):
-            cmake_args.append('-DVTKm_CUDA_Architecture=%s' %
-                              spec.variants['cuda_arch'].value)
+
+        # Source from the CudaPackage source code and VTKm CMAKE files
+        # Note that VTK-m and transitively ParaView does not support Tesla Arch
+        supported_cuda_archs = {
+            '20': 'fermi',
+            '21': 'fermi',
+            '30': 'kepler',
+            '32': 'kepler',
+            '35': 'kepler',
+            '37': 'kepler',
+            '50': 'maxwel',
+            '52': 'maxwel',
+            '53': 'maxwel',
+            '60': 'pascal',
+            '61': 'pascal',
+            '62': 'pascal',
+            '70': 'volta',
+            '72': 'volta',
+            '75': 'turing',
+            '80': 'ampere',
+            '86': 'ampere'
+        }
+
+        # If we do not specify the cuda family we can guess it by reversing
```

Comment:
Does the `unsupported` logic need to be here given the conflicts are already defined above with the variant?

Reply:
I agree, given the conflicts we should remove the unsupported architecture here.